lib/pundit.rb

```diff
@@ -71,7 +71,7 @@ def authorize(user, record, query, policy_class: nil)
 
       raise NotAuthorizedError, query: query, record: record, policy: policy unless policy.public_send(query)
 
-      record
+      record.is_a?(Array) ? record.last : record
```

Comment:
Another implementation might be `Array(record).last`, but maybe that's harder to understand?

Reply:
`Kernel.Array` can cause things like:
```ruby 
hash = { name: `Bob`, age: 40 }

Array(hash).last
#=> [:age, 40]
```

I want to reduce the possibility of modifying the record itself...
What do you think?

Reply:
Yeah you're right. I guess I have a general discomfort around typechecks, but doing some source-diving it seems like the array form is how Ruby looks up namespaces, so I guess that's just how it is 🤷‍♂